cms/djangoapps/contentstore/views/course.py

```diff
@@ -768,15 +785,31 @@ def settings_handler(request, course_key_string):
                 'about_page_editable': about_page_editable,
                 'short_description_editable': short_description_editable,
                 'upload_asset_url': upload_asset_url
-            })
+            }
+            if prerequisite_course_enabled:
+                courses, in_process_course_actions = get_courses_accessible_to_user(request)
+                # exclude current course from the list of available courses
+                courses = [course for course in courses if course.id != course_key]
+                if courses:
+                    courses = _remove_in_process_courses(courses, in_process_course_actions)
+                settings_context.update({'pre_requisite_courses': courses})
+
+            return render_to_response('settings.html', settings_context)
         elif 'application/json' in request.META.get('HTTP_ACCEPT', ''):
             if request.method == 'GET':
+                course_details = CourseDetails.fetch(course_key)
                 return JsonResponse(
-                    CourseDetails.fetch(course_key),
+                    course_details,
                     # encoder serializes dates, old locations, and instances
                     encoder=CourseSettingsEncoder
                 )
             else:  # post or put, doesn't matter.
+                # if pre-requisite course feature is enabled set pre-requisite course
+                if prerequisite_course_enabled:
+                    prerequisite_course_keys = request.json.get('pre_requisite_courses', [])
+                    if not all(is_valid_course_key(course_key) for course_key in prerequisite_course_keys):
+                        return JsonResponseBadRequest({_("error"): _("Invalid prerequisite course key")})
```

Comment:
I don't think "error" should be translated. That is a key for the user interface to retrieve the error message.